### PR TITLE
Move operator function to new style

### PR DESCRIPTION
**Pull request by jfalcou**

### CODE REVIEW

File: include/eve/module/core/regular/impl/simd/x86/roundscale.hpp [hunk 1]
```diff
@@ -16,189 +16,50 @@
...
-    case 10: return f(src, m, a, std::integral_constant<int, (10 << 4) + R>()); break;
-    case 11: return f(src, m, a, std::integral_constant<int, (11 << 4) + R>()); break;
-    case 12: return f(src, m, a, std::integral_constant<int, (12 << 4) + R>()); break;
-    case 13: return f(src, m, a, std::integral_constant<int, (13 << 4) + R>()); break;
-    case 14: return f(src, m, a, std::integral_constant<int, (14 << 4) + R>()); break;
-    case 15: return f(src, m, a, std::integral_constant<int, (15 << 4) + R>()); break;
-    default: return f(src, m, a, std::integral_constant<int, (15 << 4) + R>());
-    }
+    constexpr auto c = categorize<wide<T, N>>();
+    constexpr auto s = ((S) << 4) + rounding_mode<O>(toward_zero2);
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/simd/x86/roundscale.hpp:28]: Again, the defautl value was to_nearest, why the change ?

File: include/eve/module/core/regular/remainder.hpp [hunk 2]
```diff
@@ -7,11 +7,103 @@
...
+//!   @code
+//!   #include <eve/module/core.hpp>
+//!   @endcode
+//!
+//!   @groupheader{Callable Signatures}
+//!
+//!   @code
+//!   namespace eve
+//!   {
+//!      template< eve::signed_value T,  eve::signed_value U>
```
Comment (jfalcou) [inline include/eve/module/core/regular/remainder.hpp:53]: why signed_value while the implemenation says value ?

File: include/eve/module/core/regular/remainder.hpp [hunk 3]
```diff
@@ -7,11 +7,103 @@
...
+
+    EVE_CALLABLE_OBJECT(remainder_t, remainder_);
+  };
 
-#include <eve/module/core/regular/impl/remainder.hpp>
+//================================================================================================
+//! @addtogroup core_arithmetic
+//! @{
+//!   @var remainder
+//!   @brief Computes the  remainder after division for floating entries similarly to std::remainder
```
Comment (jfalcou) [inline include/eve/module/core/regular/remainder.hpp:39]: why speaking of floating as this consumes any value ?

File: include/eve/detail/function/operators.hpp [hunk 4]
```diff
@@ -19,63 +19,6 @@ namespace eve::detail
...
-    return a / b;
-  }
-
-  template<integral_scalar_value T>
-  EVE_FORCEINLINE  auto  rem_(EVE_SUPPORTS(cpu_), T a, T b) noexcept
-  {
-    return static_cast<T>(a % b);
-  }
 
   template<integral_value T, integral_value U>
```
[the remark this answers is not in the document]
Comment (jtlap) [reply include/eve/detail/function/operators.hpp:23]: done

File: include/eve/module/combinatorial/regular/gcd.hpp [hunk 5]
```diff
@@ -83,7 +83,7 @@ inline constexpr auto gcd = functor<gcd_t>;
         {
           while( b )
           {
-            auto r = rem(a, b);
+            auto r = fmod(a, b);
```
[the remark this answers is not in the document]
Comment (jtlap) [reply include/eve/module/combinatorial/regular/gcd.hpp:86]: it's a choice, we can change

File: include/eve/module/core/regular/core.hpp [hunk 6]
```diff
@@ -84,7 +84,6 @@
 #include <eve/module/core/regular/has_single_bit.hpp>
 #include <eve/module/core/regular/hi.hpp>
 #include <eve/module/core/regular/if_else.hpp>
-#include <eve/module/core/regular/ifnot_else.hpp>
```
[the remark this answers is not in the document]
Comment (jtlap) [reply include/eve/module/core/regular/core.hpp:87]: I removed the implementation

File: include/eve/module/core/regular/remainder.hpp (hunk 2 above)
Comment (jtlap) [reply include/eve/module/core/regular/remainder.hpp:53]: remainder needs signed value as the computed remainder can be negative
corrected

File: include/eve/module/core/regular/impl/simd/x86/roundscale.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/module/core/regular/impl/simd/x86/roundscale.hpp:28]: an error
corrected

File: include/eve/module/core/regular/impl/simd/x86/mul.hpp [hunk 7]
```diff
@@ -9,29 +9,33 @@
...
+    }
+    else
+    {
+      if      constexpr( c == category::float32x16) return _mm512_mask_mul_ps   (src, m, v, w);
+      else if constexpr( c == category::float32x8 ) return _mm256_mask_mul_ps   (src, m, v, w);
+      else if constexpr( c == category::float32x4 ) return _mm_mask_mul_ps      (src, m, v, w);
+      else if constexpr( c == category::float64x8 ) return _mm512_mask_mul_pd   (src, m, v, w);
+      else if constexpr( c == category::float64x4 ) return _mm256_mask_mul_pd   (src, m, v, w);
+      else if constexpr( c == category::float64x2 ) return _mm_mask_mul_pd      (src, m, v, w);
+      else if constexpr( match(c,category::int32x16, category::uint32x16) ) return _mm512_mask_mul_epi32(src, m, v, w);
```
[the remark this answers is not in the document]
Comment (jtlap) [reply include/eve/module/core/regular/impl/simd/x86/mul.hpp:35]: ok will check later on an avx512 computer

File: include/eve/module/core/regular/impl/simd/x86/fracscale.hpp [hunk 8]
```diff
@@ -16,189 +16,50 @@
...
-    auto          src = alternative(cx, a0, as<wide<T, N>> {});
-    auto          m   = expand_mask(cx, as<wide<T, N>> {}).storage().value;
-    constexpr int s   = ((S) << 4) + D::base_type::value;
+    if constexpr( C::is_complete) return a0;
+    else
+    {
+      constexpr auto c = categorize<wide<T, N>>();
+      auto src = alternative(cx, a0, as<wide<T, N>> {});
+      auto m   = expand_mask(cx, as<wide<T, N>> {}).storage().value;
+      constexpr auto s = ((S) << 4) + rounding_mode<O>(toward_zero2);
```
[the remark this answers is not in the document]
Comment (jtlap) [reply include/eve/module/core/regular/impl/simd/x86/fracscale.hpp:53]: an error
corrected

File: include/eve/module/core/regular/impl/simd/x86/fracscale.hpp [hunk 9]
```diff
@@ -16,189 +16,50 @@
...
-    case 10: return f(src, m, a, std::integral_constant<int, (10 << 4) + R>()); break;
-    case 11: return f(src, m, a, std::integral_constant<int, (11 << 4) + R>()); break;
-    case 12: return f(src, m, a, std::integral_constant<int, (12 << 4) + R>()); break;
-    case 13: return f(src, m, a, std::integral_constant<int, (13 << 4) + R>()); break;
-    case 14: return f(src, m, a, std::integral_constant<int, (14 << 4) + R>()); break;
-    case 15: return f(src, m, a, std::integral_constant<int, (15 << 4) + R>()); break;
-    default: return f(src, m, a, std::integral_constant<int, (15 << 4) + R>());
-    }
+    constexpr auto c = categorize<wide<T, N>>();
+    constexpr auto s = ((S) << 4) + rounding_mode<O>(toward_zero2);
```
[the remark this answers is not in the document]
Comment (jtlap) [reply include/eve/module/core/regular/impl/simd/x86/fracscale.hpp:28]: an error
corrected

File: include/eve/module/core/regular/remainder.hpp (hunk 3 above)
Comment (jtlap) [reply include/eve/module/core/regular/remainder.hpp:39]: because std::remainder accepts integers only from c++23 and in this case return double.
I am just meaning that it acts as std for floating

**Comment by DenisYaroshevskiy** [comment]
Maybe merge this ona and start the next one? It's enormous
